ClientKit/ClientReportTypes.cs

```diff
@@ -27,37 +36,81 @@ namespace ClientKit
         [StructLayout(LayoutKind.Sequential)]
         public struct PositionReport
         {
-            public Int32 sensor;
-            public Vec3 xyz;
+            public ChannelCount sensor;
+            public PositionState xyz;
         }
 
         [StructLayout(LayoutKind.Sequential)]
         public struct OrientationReport
         {
-            public Int32 sensor;
+            public ChannelCount sensor;
             public Quaternion rotation;
         }
 
         [StructLayout(LayoutKind.Sequential)]
         public struct PoseReport
         {
-            public Int32 sensor;
+            public ChannelCount sensor;
             public Pose3 pose;
         }
 
         [StructLayout(LayoutKind.Sequential)]
         public struct ButtonReport
         {
-            public Int32 sensor;
+            public ChannelCount sensor;
             public Byte state;
         }
 
         [StructLayout(LayoutKind.Sequential)]
         public struct AnalogReport
         {
-            public Int32 sensor;
+            public ChannelCount sensor;
             public Double state;
         }
+
+        [StructLayout(LayoutKind.Sequential)]
+        public struct Location2DReport
+        {
+            public ChannelCount sensor;
+            public Location2DState xy;
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        public struct DirectionReport
+        {
+            public ChannelCount sensor;
+            public DirectionState direction;
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        public struct EyeTracker3DState
+        {
+            public bool directionValid;

```

Comment:
OK, ugh, not worth it. Users can just look at the bools like everybody else then. Was just curious since we'll be adding a new tracker callback type that has bools as well, to cleanly handle the case where an app wants full pose if it can get it, otherwise it's ok with some subset - the so-called "subset message" problem.